Pod/Extensions/SSO/SKYAuthContainer+SSO.m

```diff
@@ -161,4 +164,28 @@ - (NSURL *)sso_genCallbackURL:(NSString *)scheme
     return [[NSURL alloc] initWithScheme:scheme host:host path:@"/auth_handler"];
 }
 
+#pragma mark - Custom Token
+
+- (void)loginWithCustomToken:(NSString *)customToken
+           completionHandler:(SKYContainerUserOperationActionCompletion)completionHandler
+{
+    SKYLoginCustomTokenOperation *op =
+        [SKYLoginCustomTokenOperation operationWithCustomToken:customToken];
+    op.loginCompletionBlock = ^(SKYRecord *user, SKYAccessToken *accessToken, NSError *error) {
+        if (error) {
+            if (completionHandler) {
+                completionHandler(user, error);
+            }
+            return;
+        }
+
+        [self.container.auth updateWithUser:user accessToken:accessToken];
+
+        if (completionHandler) {
+            completionHandler(user, error);
+        }
```

Comment:
@cheungpat I just tried the api and do some ui operation in the `completionHandler` and got error `UI API called on a background thread`. I think we also need to call the `completionHandler` in main thread like the other login operation? Both error and success handler.

https://github.com/SkygearIO/skygear-SDK-iOS/blob/54b3f0d0475f3b14290842a2ffd5c1aca7111bb7/Pod/Classes/SKYAuthContainer.m#L94

thanks kuen kuen!!

Reply:
right